python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -536,6 +538,37 @@ std::string getQIRLL(const std::string &name, MlirModule module,
   return str;
 }
 
+std::string getASM(const std::string &name, MlirModule module,
+                   cudaq::OpaqueArguments &runtimeArgs) {
+  ScopedTraceWithContext(cudaq::TIMING_JIT, "getASM", name);
+  auto noneType = mlir::NoneType::get(unwrap(module).getContext());
+
+  auto [jit, rawArgs, size, returnOffset] =
+      jitAndCreateArgs(name, module, runtimeArgs, {}, noneType);
+  auto cloned = unwrap(module).clone();
+  auto context = cloned.getContext();
+
+  PassManager pm(context);
+  pm.addPass(cudaq::opt::createLambdaLiftingPass());
+  cudaq::opt::addPipelineTranslateToOpenQASM(pm);
+
+  mlir::DefaultTimingManager tm;
+  tm.setEnabled(cudaq::isTimingTagEnabled(cudaq::TIMING_JIT_PASSES));
+  auto timingScope = tm.getRootScope(); // starts the timer
+  pm.enableTiming(timingScope);         // do this right before pm.run
+  if (failed(pm.run(cloned)))
+    throw std::runtime_error("getASM: code generation failed.");
+  timingScope.stop();
```

Comment:
I think that running timing on this code path probably isn't necessary and therefore this could be simplified a little bit:
```suggestion
  if (failed(pm.run(cloned)))
    throw std::runtime_error("getASM: code generation failed.");
```